ultra/ultra/baselines/common/baseline_state_preprocessor.py

```diff
@@ -0,0 +1,200 @@
+# MIT License
+#
+# Copyright (C) 2021. Huawei Technologies Co., Ltd. All rights reserved.
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+import collections.abc
+import numpy as np
+import torch
+
+from ultra.baselines.common.state_preprocessor import StatePreprocessor
+from ultra.utils.common import rotate2d_vector
+
+
+class BaselineStatePreprocessor(StatePreprocessor):
+    """The State Preprocessor used by the baseline agents."""
+
+    _NUM_LOOKAHEAD = 100
```

Comment:
Why do we need this hard-coded value? I know it was there before, but I think it's used only in extract_closest_waypoint and it already defaults to 100. Can we always use observation_num_lookahead?

Reply:
Ahh good call. I will get rid of it. And good question, @adreena can you comment on this? Can we just always use the `observation_num_lookahead` in `params.yaml` for the `num_lookahead` parameter when we call `extract_closest_waypoint(...)`? It is done here: https://github.com/huawei-noah/SMARTS/blob/c8473049b4b52ad3eb29f7c7eb93753c2d70df55/ultra/ultra/baselines/common/state_preprocessor.py#L102 But not here: https://github.com/huawei-noah/SMARTS/blob/c8473049b4b52ad3eb29f7c7eb93753c2d70df55/ultra/ultra/baselines/adapter.py#L58 was there any reason why it was set to 100?